packages/@aws-cdk/aws-codepipeline-actions/lib/codebuild/build-action.ts

```diff
@@ -60,6 +60,21 @@ export interface CodeBuildActionProps extends codepipeline.CommonAwsActionProps
    * @default CodeBuildActionType.BUILD
    */
   readonly type?: CodeBuildActionType;
+
+  /**
+   * Whether to validate if the action,
+   * if it references a project in a different account,
+   * does not have any outputs.
+   * This is a known CodeBuild limitation.
+   * If the action is cross-account, and has outputs,
+   * and this property is true (which is the default),
+   * an exception will be thrown.
+   * You can set this to false to skip this validation.
+   *
+   * @default true
+   * @see https://github.com/aws/aws-cdk/issues/4169
+   */
+  readonly validateCrossAccountOutputs?: boolean;
 }
```

Comment:
What’s the benefit of having this option if it doesn’t work?

Reply:
At some point, CodeBuild will fix this behavior, and this escape hatch allows the customer to override our check without having to upgrade their version of the CDK.

Reply:
Will their fix block CDK users? Seems like this will still work even after this will be fixed by CodeBuild, and then we can leisurely remove this workaround without breaking users, no?

Reply:
Sorry I thought this was about the encryption key workaround and not the output artifacts. Still... I prefer that we didn’t pollute our API with this because after the issue is fixed, we will stay with an api that doesn’t have a meaning. Let’s not expose this for now. We update the cdk every week. When this issue is resolved by codebuild we can easily remove this fix and release an update.

Reply:
After the issue is fixed, this is one deprecated property that will simply not be used anywhere. In my opinion, the cost for us is pretty much zero.

Reply:
Let's not future proof. When CodeBuild fixes the issue we can remove the workaround.

Reply:
Removed `validateCrossAccountOutputs`.

Reply:
I just thought of an edge-case with this.

If the subsequent action that reads the output of the CodeBuild action is in the same account as the CodeBuild project, the artifact will be read correctly (even though it can't be read by anything from the pipeline account, because of the wrong key used to encrypt this).

Does this possibly change your stance on having an escape hatch with `validateCrossAccountOutputs` @eladb ?

Reply:
Let's leave it out for now and add if users will request this capability. Then we can think what's the best way to expose it. OK?

Reply:
OK. I'm merging this in then.